lib/rspec/openapi/record_builder.rb

```diff
@@ -99,21 +107,18 @@ def rack_test?(context)
   end
 
   # @param [ActionDispatch::Request] request
-  def find_rails_route(request, app: Rails.application, fix_path: true)
-    # Reverse the destructive modification by Rails https://github.com/rails/rails/blob/v6.0.3.4/actionpack/lib/action_dispatch/journey/router.rb#L33-L41
-    if fix_path && !request.script_name.empty?
-      request = request.dup
-      request.path_info = File.join(request.script_name, request.path_info)
-    end
-
+  def find_rails_route(request, app: Rails.application, path_prefix: "")
     app.routes.router.recognize(request) do |route|
+      path = route.path.spec.to_s
       if route.app.matches?(request)
-        return find_rails_route(request, app: route.app.app, fix_path: false) if route.app.engine?
-
-        return route
+        if route.app.engine?
+          route, path = find_rails_route(request, app: route.app.app, path_prefix: path)
+          next if route.nil?
+        end
+        return [route, path_prefix + path]
       end
     end
-    raise "No route matched for #{request.request_method} #{request.path_info}"
+    nil
```

Comment:
Q) Why this error message is removed?

Reply:
When two engines are mounted on the same root path, and the first doesn't resolve, the resolution needs to continue to the next engine. The raise was moved to the caller, when all engines are resolved